aztec_macros/src/lib.rs

```diff
@@ -274,18 +299,20 @@ fn is_custom_attribute(attr: &SecondaryAttribute, attribute_name: &str) -> bool
 fn transform_module(
     module: &mut SortedModule,
     crate_id: &CrateId,
-    context: &Context,
-) -> Result<bool, (DefCollectorErrorKind, FileId)> {
+    context: &HirContext,
+) -> Result<bool, (MacroError, FileId)> {
     let mut has_transformed_module = false;
 
     // Check for a user defined storage struct
-    let storage_defined = check_for_storage_definition(&module);
+    let storage_defined = check_for_storage_definition(module);
 
-    if storage_defined && !check_for_compute_note_hash_and_nullifier_definition(&module) {
+    if storage_defined && !check_for_compute_note_hash_and_nullifier_definition(module) {
         let crate_graph = &context.crate_graph[crate_id];
         return Err((
-            DefCollectorErrorKind::AztecComputeNoteHashAndNullifierNotFound {
-                span: Span::default(), // Add a default span so we know which contract file the error originates from
+            MacroError {
+                span: Some(Span::default()), // Add a default span so we know which contract file the error originates from
+                primary_message: "compute_note_hash_and_nullifier function not found. Define it in your contract.".to_owned(),
+                secondary_message: None,
```

Comment:
Maybe I misunderstand -- the aztec macro has a `AztecMacroError` enum with the variants. Did you mean to have a variant like that somewhere else?

Reply:
Should we not then declare an `AztecMacroError` and call into rather than declaring a new `MacroError` here?

Reply:
The idea was that the frontend no longer knew about the specifc macro -- so theoretically, we could add another macro here to manipulate the AST and it would not break the frontend

Reply:
Note tht MacroError is defined in the frontend/compiler and in this aztec_macros file we have a AztecMacroError. AztecMacroError gets converted to MacroError for the frontend in the highlighted lines

Reply:
^ The above is true, but it was not being used. I have changed it now